lib/rest-adapter.js

```diff
@@ -256,12 +257,27 @@ RestAdapter.prototype.createHandler = function() {
 
   var handleUnknownPaths = this._shouldHandleUnknownPaths();
 
+  // mount rest route handler
+  var customRouter = new RestRouter();
+  root.use(customRouter.handle());
+  
   classes.forEach(function(restClass) {
     var router = express.Router();
     var className = restClass.sharedClass.name;
 
     debug('registering REST handler for class %j', className);
-
+    
+    // Mount the remoteClass router on all class routes.
+    restClass
+      .routes
+      .forEach(function (route) {
+        debug('    at %s', route.path);
+        root.use(route.path, router);
+        route.fullPath = restClass.getPath(); //add fullPath for Trie insertion
```

Comment:
This is smelly. Surely the full route of the path is not just the path of the class? Perhaps `route.classPath` would be a better property name?